src/coreclr/jit/fginline.cpp

```diff
@@ -653,7 +653,12 @@ class SubstitutePlaceholdersAndDevirtualizeWalker : public GenTreeVisitor<Substi
                         }
 
                         call->GetSingleInlineCandidateInfo()->exactContextHandle = context;
-                        INDEBUG(call->GetSingleInlineCandidateInfo()->inlinersContext = call->gtInlineContext);
+#ifdef DEBUG
+                        // Update inline context for Debug.
+                        call->gtInlineContext = call->GetSingleInlineCandidateInfo()->inlinersContext;
+                        DebugInfo debugInfo(call->gtInlineContext, ILLocation(call->gtRawILOffset, false, true));
+                        m_curStmt->SetDebugInfo(debugInfo);
+#endif
```

Comment:
This cannot be setting debug-info under DEBUG only. Debug information needs to be the same for release and debug JITs.